test/conformance/testing/include/uur/fixtures.h

```diff
@@ -75,168 +120,39 @@ struct urAllDevicesTest : urPlatformTest {
-            return uur::GetPlatformAndDeviceName(info.param);                  \
-        })
-
-namespace uur {
-
-template <class T>
-struct urDeviceTestWithParam
+template <typename... Args>
+struct urDeviceTest
     : urPlatformTest,
-      ::testing::WithParamInterface<std::tuple<ur_device_handle_t, T>> {
+      ::testing::WithParamInterface<std::tuple<ur_device_handle_t, Args...>> {
     void SetUp() override {
         UUR_RETURN_ON_FATAL_FAILURE(urPlatformTest::SetUp());
         device = std::get<0>(this->GetParam());
     }
     // TODO - I don't like the confusion with GetParam();
-    const T &getParam() const { return std::get<1>(this->GetParam()); }
-    ur_device_handle_t device;
-};
-
-struct urContextTest : urDeviceTest {
-    void SetUp() override {
-        UUR_RETURN_ON_FATAL_FAILURE(urDeviceTest::SetUp());
-        ASSERT_SUCCESS(urContextCreate(1, &device, nullptr, &context));
-        ASSERT_NE(context, nullptr);
-    }
-
-    void TearDown() override {
-        EXPECT_SUCCESS(urContextRelease(context));
-        UUR_RETURN_ON_FATAL_FAILURE(urDeviceTest::TearDown());
-    }
-
-    ur_context_handle_t context = nullptr;
-};
-
-struct urSamplerTest : urContextTest {
-    void SetUp() override {
-        UUR_RETURN_ON_FATAL_FAILURE(urContextTest::SetUp());
-        sampler_desc = {
-            UR_STRUCTURE_TYPE_SAMPLER_DESC,   /* stype */
-            nullptr,                          /* pNext */
-            true,                             /* normalizedCoords */
-            UR_SAMPLER_ADDRESSING_MODE_CLAMP, /* addressing mode */
-            UR_SAMPLER_FILTER_MODE_LINEAR,    /* filterMode */
-        };
-        ASSERT_SUCCESS(urSamplerCreate(context, &sampler_desc, &sampler));
-    }
-
-    void TearDown() override {
-        EXPECT_SUCCESS(urSamplerRelease(sampler));
-        UUR_RETURN_ON_FATAL_FAILURE(urContextTest::TearDown());
-    }
-
-    ur_sampler_handle_t sampler;
-    ur_sampler_desc_t sampler_desc;
-};
-
-struct urMemBufferTest : urContextTest {
-    void SetUp() override {
-        UUR_RETURN_ON_FATAL_FAILURE(urContextTest::SetUp());
-        ASSERT_SUCCESS(urMemBufferCreate(context, UR_MEM_FLAG_READ_WRITE, 4096,
-                                         nullptr, &buffer));
-        ASSERT_NE(nullptr, buffer);
-    }
-
-    void TearDown() override {
-        if (buffer) {
-            EXPECT_SUCCESS(urMemRelease(buffer));
-        }
-        UUR_RETURN_ON_FATAL_FAILURE(urContextTest::TearDown());
-    }
-
-    ur_mem_handle_t buffer = nullptr;
-};
-
-struct urMemImageTest : urContextTest {
-    void SetUp() override {
-        UUR_RETURN_ON_FATAL_FAILURE(urContextTest::SetUp());
-    }
-
-    void TearDown() override {
-        if (image) {
-            EXPECT_SUCCESS(urMemRelease(image));
-        }
-        UUR_RETURN_ON_FATAL_FAILURE(urContextTest::TearDown());
+    template <size_t Index = 1> auto &getParam() const {
+        return std::get<Index>(this->GetParam());
```

Comment:
There is an old `TODO` here about `getParam` and `GetParam` being very similar. I spent some time thinking about it and still don't know what else to name it. Open to suggestions :)